gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -698,6 +742,44 @@ MM_MemoryPoolAddressOrderedList::internalAllocateTLH(MM_EnvironmentBase *env, ui
 	return false;
 }
 
+uintptr_t
+MM_MemoryPoolAddressOrderedList::determineConsumedSizeForTLH(MM_EnvironmentBase *env, MM_HeapLinkedFreeHeader* freeEntry, uintptr_t maximumSizeInBytesRequired) {
+	bool const compressed = compressObjectReferences();
+
+	/* Consume the bytes and set the return pointer values */
+	uintptr_t freeEntrySize = freeEntry->getSize();
+	Assert_MM_true(freeEntrySize >= _minimumFreeEntrySize);
+	uintptr_t consumedSize = (maximumSizeInBytesRequired > freeEntrySize) ? freeEntrySize : maximumSizeInBytesRequired;
+
+	if (isAlignmentForParallelGCRequired()) {
+		if (!alignTLHForParallelGC(env, freeEntry, &consumedSize)) {
+			/* If alignment was required and it failed (i.e resulted in consumedSize < minEntrySize), abandon entry and retry */
+			abandonHeapChunk((void *)freeEntry, (void *)(((uintptr_t)freeEntry) + freeEntrySize));
+
+			_freeMemorySize -= freeEntrySize;
+			_allocDiscardedBytes += freeEntrySize;
+
+			MM_HeapLinkedFreeHeader *entryNext = freeEntry->getNext(compressed);
+
+			updatePrevCardUnalignedFreeEntry(entryNext, FREE_ENTRY_END);
+
+			_heapFreeList = entryNext;
+			_freeEntryCount -= 1;
+
+			consumedSize = 0;
+		}
+	}
+
+	return consumedSize;
+}
+
+void
+MM_MemoryPoolAddressOrderedList::updatePrevCardUnalignedFreeEntry(MM_HeapLinkedFreeHeader* entryNext, MM_HeapLinkedFreeHeader* value) {
+	if (entryNext == _firstCardUnalignedFreeEntry) {
+		_prevCardUnalignedFreeEntry = value;
+	}
+}
```

Comment:
move it to *hpp and inline it
there are couple of spots in non-TLH path that this can be used too